feat: caip-25 example for 5792 capabilities (#400)
reown-com/reown-docs: walletkit/android/eip5792.mdx

  "expiry": "2022-12-24T17:07:31+00:00",
  "caip154": {
    "supported": "true"
  },
  "flow-control": {
    "loose": ["halt", "continue"],
    "strict": ["continue"]
  },
  "atomic": {
    "status": "ready"
  }
},
"scopedProperties": {
  "eip155:1": {
    "atomic": {
      "status": "supported"
    }
  },
  "eip155:137": {
    "atomic": {
      "status": "unsupported"
    }
  },
  "eip155:84532": {
    "eip155:83532:0x0910e12C68d02B561a34569E1367c9AAb42bd810": {
      "auxiliaryFunds": {
        "supported": false
      },
      "atomic": {
        "status": "supported"
      }
    }
  }
}
```
- Capabilities shared across all address in a namespace can be expressed at top-level
- Address-specific capabilities can include exceptions to scope-wide capabilities

### Atomic Capability

According to [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792), the `atomic` capability specifies how the wallet handles batches of transactions. It has three possible values:

- `supported` — The wallet executes calls atomically and contiguously.
- `ready` — The wallet can upgrade to support atomic execution, pending user approval.
- `unsupported` — The wallet provides no atomicity guarantees.

> This capability is expressed per chain and is crucial for determining how `wallet_sendCalls` with `atomicRequired: true` will be handled.

Reviewer: i wonder if we need to cover it here 🤔

Author: i think it doesn't harm to have that

- `wallet_getCallsStatus`: Returns the status of a call batch that was sent via wallet_sendCalls.
- `wallet_showCallsStatus`: Requests that a wallet shows information about a given call bundle that was sent with wallet_sendCalls.
- `wallet_getCapabilities`: This RPC allows an application to request capabilities from a wallet (e.g. batch transactions, paymaster communication).

> ## Usage
> 
> ### wallet_getCapabilities Example
> 
> The `wallet_getCapabilities` method is used to request information about what capabilities a wallet supports. Following EIP-5792, here's how it should be implemented:

Reviewer 2: ```suggestion
## Usage

### wallet_getCapabilities Example

The `wallet_getCapabilities` method is used to request information about the capabilities supported by a wallet. In accordance with EIP-5792, here’s how it should be implemented:
```

Minor grammar fix

### Request
```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "wallet_getCapabilities",
  "params": []
}
```

### Response

> The wallet should return a response following EIP-5792, where capabilities are organized by chain ID:

Reviewer 2: ```suggestion
The wallet should return a response compliant with EIP-5792, where capabilities are organized by chain ID:
```

Grammar fix

```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "result": {
    "0x2105": {
      "atomic": {
        "status": "supported"
      }
    },
    "0x14A34": {
      "atomic": {
        "status": "unsupported"
      }
    }
  }
}
```

## Capabilities in CAIP-25 Connection Requests

CAIP-25 defines how capabilities can be expressed in wallet-to-dapp connections. These capabilities control how methods like `wallet_sendCalls` behave.

### Session Properties

> In a connection request, dapps can request capabilities via `sessionProperties`. These can be universal (across all chains) or chain-specific:

Reviewer 2: ```suggestion
In a connection request, dApps can request capabilities through `sessionProperties`. These capabilities can be universal (applying to all chains) or chain-specific:
```

```json
"sessionProperties": {
  "expiry": "2022-12-24T17:07:31+00:00",
  // Universal capabilities (across all chains)
  "caip154": {
    "supported": "true"
  },
  "flow-control": {
    "loose": [], 
    "strict": [],

>     "exoticThirdThing": [] // Unrecognized config will be dropped by wallet

Author: Yes, makes sense

  },
  "atomic": {
    "status": "supported"
  }
}
```

### Scoped Properties

For chain-specific capabilities, dapps use `scopedProperties`:

```json
"scopedProperties": {
  // Chain-specific requests
  "eip155:8453": {
    "paymasterService": {
      "supported": true
    },
    "sessionKeys": {
      "supported": true
    }
  },
  "eip155:84532": {
    "auxiliaryFunds": {
      "supported": true
    }
  }
}
```

### Wallet Response

> A wallet's response should indicate which capabilities it actually supports, following EIP-5792 and CAIP-25:

Reviewer 2: ```suggestion
The wallet’s response should specify the capabilities it supports, in accordance with EIP-5792 and CAIP-25:
```

Grammar fix

```json
"sessionProperties": {
  "expiry": "2022-12-24T17:07:31+00:00",
  "caip154": {
    "supported": "true"
  },
  "flow-control": {
    "loose": ["halt", "continue"],
    "strict": ["continue"]
  },
  "atomic": {
    "status": "ready"
  }
},
"scopedProperties": {
  "eip155:1": {
    // Capabilities shared across all addresses in this namespace
    "atomic": {
      "status": "supported"
    }
  },
  "eip155:137": {
    "atomic": {
      "status": "unsupported"
    }
  },
  "eip155:84532": {
    // Address-specific capabilities can include exceptions to scope-wide capabilities
    "eip155:83532:0x0910e12C68d02B561a34569E1367c9AAb42bd810": {
      "auxiliaryFunds": {
        "supported": false
      },
      "atomic": {
        "status": "supported"
      }
    }
    // No need to declare implicit/default capabilities per-address
  }
}
```

> ### Atomic Capability
> 
> According to EIP-5792, the `atomic` capability specifies how the wallet will execute batches of transactions. It has three possible values:
> 
> - `supported` - The wallet will execute calls atomically and contiguously
> - `ready` - The wallet can upgrade to support atomic execution pending user approval
> - `unsupported` - The wallet provides no atomicity guarantees
> 
> This capability is expressed per chain and is crucial for determining how `wallet_sendCalls` with `atomicRequired: true` will be handled.

Reviewer 2: ```suggestion
### Atomic Capability

According to [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792), the `atomic` capability specifies how the wallet handles batches of transactions. It has three possible values:

- `supported` — The wallet executes calls atomically and contiguously.
- `ready` — The wallet can upgrade to support atomic execution, pending user approval.
- `unsupported` — The wallet provides no atomicity guarantees.

This capability is expressed per chain and is crucial for determining how `wallet_sendCalls` with `atomicRequired: true` will be handled.
```

Grammar fix
